CardanoSharp.Wallet/KeyService.cs

```diff
@@ -25,20 +25,23 @@ public interface IKeyService
     public class KeyService : IKeyService
     {
         #region BIP39
-        private readonly int[] allowedEntropyLengths = { 16, 20, 24, 28, 32 };
-        private static readonly int[] allowedWordLengths = { 12, 15, 18, 21, 24 };
+        private readonly int[] allowedEntropyLengths = { 12, 16, 20, 24, 28, 32 };
+        private static readonly int[] allowedWordLengths = { 9, 12, 15, 18, 21, 24 };
         private uint[] wordIndexes;
         private string[] allWords;
 
         public string Generate(int wordSize, WordLists wl = WordLists.English)
         {
-            var entropySize = allowedEntropyLengths[allowedWordLengths.ToList().FindIndex(x => x == wordSize)];
+            if (!allowedWordLengths.Contains(wordSize))
+                throw new ArgumentOutOfRangeException(nameof(wordSize), $"{nameof(wordSize)} must be one of the following values ({string.Join(", ", allowedWordLengths)})");
+
+            var entropySize = allowedEntropyLengths[Array.FindIndex(allowedWordLengths, x => x == wordSize)];
+            if (!allowedEntropyLengths.Contains(entropySize))
+                throw new ArgumentOutOfRangeException(nameof(entropySize), $"Derived entropy {entropySize} is not within the allowed values ({string.Join(", ", allowedEntropyLengths)})");
 
             var rng = new RNGCryptoServiceProvider();
             if (rng is null)
```

Comment:
Unless I am overlooking something in the way RNGCryptoServiceProvider is initialised, the condition if (rng is null) should never be true?

Reply:
yeah i believe thats an old condition, we can probably remove that